tests/SCsub

```diff
@@ -13,3 +13,9 @@ env_tests.add_source_files(env.tests_sources, "*.cpp")
 
 lib = env_tests.add_library("tests", env.tests_sources)
 env.Prepend(LIBS=[lib])
+
+# we must disable the THREAD_LOCAL entirely in doctest to prevent crashes on debugging
+# since we link with /MT thread_local is always expired when the header is used, so the debugger crashes the engine and it causes weird errors
+# explained in https://github.com/onqtam/doctest/issues/401
+if env_tests["platform"] == "windows":
+    env_tests.Append(CPPDEFINES=["DOCTEST_THREAD_LOCAL"])
```

Comment:
Move this up with the other define. It might work as is but it's clearer to set up defines before building the library ;)

Also capitalize your comments, and keep line lengths to a sensible max (it's weird that black didn't fail though...).

Reply:
done :) and fixed